app/controllers/incomeSources/cease/CeaseCheckIncomeSourceDetailsController.scala

```diff
@@ -0,0 +1,242 @@
+    (checkSessionTimeout andThen authenticate andThen retrieveNino andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+      implicit user =>
+        handleRequest(
+          sources = user.incomeSources,
+          isAgent = false,
+          None,
+          incomeSourceType = incomeSourceType
+        )
+    }
+
+  def showAgent(incomeSourceType: IncomeSourceType): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap {
+          implicit mtdItUser =>
+            handleRequest(
+              sources = mtdItUser.incomeSources,
+              isAgent = true,
+              incomeSourceType = incomeSourceType
+            )
+        }
+  }
+
+  def handleSubmitRequest(isAgent: Boolean, incomeSourceType: IncomeSourceType)(implicit user: MtdItUser[_], request: Request[_]): Future[Result] = withIncomeSourcesFS {
+
+    val redirectAction: Call = (isAgent, incomeSourceType) match {
+      case (true, SelfEmployment) => routes.IncomeSourceCeasedObligationsController.showAgent(SelfEmployment)
+      case (false, SelfEmployment) => routes.IncomeSourceCeasedObligationsController.show(SelfEmployment)
+      case (true, UkProperty) => routes.IncomeSourceCeasedObligationsController.showAgent(UkProperty)
+      case (false, UkProperty) => routes.IncomeSourceCeasedObligationsController.show(UkProperty)
+      case (true, ForeignProperty) => routes.IncomeSourceCeasedObligationsController.showAgent(ForeignProperty)
+      case (false, ForeignProperty) => routes.IncomeSourceCeasedObligationsController.show(ForeignProperty)
+      case _ => routes.IncomeSourceNotCeasedController.show(isAgent, incomeSourceType)
+    }
+
+    val sessionDataFuture = getSessionData(incomeSourceType)
+
+    sessionDataFuture.flatMap {
+      case (Right(Some(incomeSourceId)), Right(Some(cessationEndDate))) =>
+        updateIncomeSourceservice
+          .updateCessationDate(user.nino, incomeSourceId, cessationEndDate).flatMap {
+          case Right(_) =>
+            Future.successful(Redirect(redirectAction))
+          case _ =>
+            Logger("application").error(s"[CheckCeaseBusinessDetailsController][handleSubmitRequest]:" +
+              s" Unsuccessful update response received")
+            Future.successful {
+              Redirect(controllers.incomeSources.cease.routes.IncomeSourceNotCeasedController.show(isAgent, SelfEmployment))
+            }
+        }
+
+      case (Right(None), Right(Some(cessationEndDate))) =>
+        val propertyIncomeSources = (incomeSourceType) match {
+          case (UkProperty) => user.incomeSources.properties.find(x => x.isUkProperty && !x.isCeased)
+          case (ForeignProperty) => user.incomeSources.properties.find(x => x.isForeignProperty && !x.isCeased)
+          case (SelfEmployment) => None
+        }
+        if(propertyIncomeSources.isEmpty){
```

Comment:
Should fix:
Can you format this file? 🙏

Reply:
I've just pushed auto format, lmk it needs further